gwctl/pkg/utils/types.go

```diff
@@ -30,22 +28,61 @@ import (
 	"sigs.k8s.io/gateway-api/gwctl/pkg/policymanager"
 )
 
-type CmdParams struct {
-	K8sClients    *common.K8sClients
-	PolicyManager *policymanager.PolicyManager
-	Out           io.Writer
+type Factory interface {
+	K8sClients() (*common.K8sClients, error)
+	PolicyManager() (*policymanager.PolicyManager, error)
 }
```

Comment:
Why does this need to be an interface? Are you expecting other implementations of the interface to be added in the future? (Not saying this is necessarily bad, just trying to understand the use case).

Reply:
Nice question. Added an _internal_ comment for future context as well. Happy to explain it differently if needed.